source/geod24/Registry.d

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+
+    Registry implementation for multi-threaded access
+
+    This registry allows to look up a `Tid` based on a `string`.
+    It is extracted from the `std.concurrency` module to make it reusable
+
+*******************************************************************************/
+
+module geod24.Registry;
+
+import core.sync.mutex;
+import geod24.concurrency;
+
+/// Ditto
+public shared struct Registry
```

Comment:
What are the semantics of a `shared` struct? I've heard of `synchronized` classes, but never `shared` aggregates.

Reply:
Ah let me guess, everything inside is like `shared:` for fields.

Reply:
I wonder if you can declare it as `shared synchronized struct`, I think this would make all methods lock on the built-in mutex. Concurrency keyword soup, haha.

Reply:
(but actually don't do that, it's better to be explicit IMO)

Reply:
You can't have `synchronized` on `struct` anyway, it's only for classes.

Reply:
So there is nothing to do here, right ?